src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2390,6 +2403,444 @@ def remove_triggered_webjob(cmd, resource_group_name, name, webjob_name, slot=No
+    return logger.warning("not a webapp")
+
+
+def functionapp_list_hc(cmd, name, resource_group, slot=None):
+    webapp = _generic_site_operation(cmd.cli_ctx, resource_group, name, 'get', slot)
+    linux_webapp = show_webapp(cmd, resource_group, name, slot)
+    is_linux = linux_webapp.reserved
+
+    if webapp.kind == "functionapp":
+        return list_hc(cmd, name, resource_group, slot)
+    if is_linux:
+        return logger.warning("hybrid connections not supported on a linux app")
+    return logger.warning("not a functionapp")
+
+
+def list_hc(cmd, name, resource_group, slot):
+    client = web_client_factory(cmd.cli_ctx)
+    if slot is None:
+        listed_vals = client.web_apps.list_hybrid_connections(resource_group, name)
+    else:
+        listed_vals = client.web_apps.list_hybrid_connections_slot(resource_group, name, slot)
+
+    # reformats hybrid connection, to prune unnecessary fields
+    mod_list = []
+    for x in listed_vals.additional_properties["value"]:
+        properties = x["properties"]
+        resourceGroup = x["id"].split("/")
+        mod_hc = {
+            "id": x["id"],
+            "location": x["location"],
+            "name": x["name"],
+            "properties": {
+                "hostname": properties["hostname"],
+                "port": properties["port"],
+                "relayArmUri": properties["relayArmUri"],
+                "relayName": properties["relayName"],
+                "serviceBusNamespace": properties["serviceBusNamespace"],
+                "serviceBusSuffix": properties["serviceBusSuffix"]
+            },
+            "resourceGroup": resourceGroup[4],
+            "type": x["type"]
+        }
+        mod_list.append(mod_hc)
+    return mod_list
+
+
+def webapp_add_hc(cmd, name, resource_group, namespace, hybrid_connection, slot=None):
+    webapp = _generic_site_operation(cmd.cli_ctx, resource_group, name, 'get', slot)
+    linux_webapp = show_webapp(cmd, resource_group, name, slot)
+    is_linux = linux_webapp.reserved
+
+    if webapp.kind == "app":
+        return add_hc(cmd, name, resource_group, namespace, hybrid_connection, slot)
+    if is_linux:
+        return logger.warning("hybrid connections not supported on a linux app")
+    return logger.warning("not a webapp")
+
+
+def functionapp_add_hc(cmd, name, resource_group, namespace, hybrid_connection, slot=None):
+    webapp = _generic_site_operation(cmd.cli_ctx, resource_group, name, 'get', slot)
+    linux_webapp = show_webapp(cmd, resource_group, name, slot)
+    is_linux = linux_webapp.reserved
+
+    if webapp.kind == "functionapp":
+        return add_hc(cmd, name, resource_group, namespace, hybrid_connection, slot)
+    if is_linux:
+        return logger.warning("hybrid connections not supported on a linux app")
+    return logger.warning("not a functionapp")
+
+
+def add_hc(cmd, name, resource_group, namespace, hybrid_connection, slot):
```

Comment:
Given this a helper method that is not directly called from a command - make this _add_hc (private method)